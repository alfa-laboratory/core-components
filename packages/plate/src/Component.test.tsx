import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { CheckmarkCircleMIcon } from '@alfalab/icons-glyph';

import { Plate } from './index';

describe('Plate', () => {
    describe('Snapshots tests', () => {
        it('should match snapshot', () => {
            const { container } = render(
                <Plate view='positive' title='title' leftAddons={<CheckmarkCircleMIcon />}>
                    text
                </Plate>,
            );

            expect(container).toMatchSnapshot();
        });
    });

    it('should forward ref', () => {
        const ref = jest.fn();
        const dataTestId = 'test-id';
        const { getByTestId } = render(<Plate ref={ref} dataTestId={dataTestId} />);

        expect(ref.mock.calls).toEqual([[getByTestId(dataTestId)]]);
    });

    it('should set `data-test-id` attribute', () => {
        const dataTestId = 'test-id';
        const { getByTestId } = render(<Plate dataTestId={dataTestId} />);

        expect(getByTestId(dataTestId).tagName).toBe('DIV');
    });

    describe('Classes tests', () => {
        it('should set `className` class', () => {
            const className = 'test-class';
            const { container } = render(<Plate className={className} />);

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should set `positive` class if `view` prop is `positive`', () => {
            const view = 'positive';
            const { container } = render(<Plate view={view} />);

            expect(container.firstElementChild).toHaveClass(view);
        });

        it('should set `hasCloser` class', () => {
            const { container } = render(<Plate hasCloser={true} />);

            expect(container.firstElementChild).toHaveClass('hasCloser');
        });

        it('should set `foldable` class', () => {
            const { container } = render(
                <Plate foldable={true} title='title'>
                    text
                </Plate>,
            );

            expect(container.firstElementChild).toHaveClass('foldable');
        });

        it('should not set `foldable` class with empty `title`', () => {
            const { container } = render(<Plate foldable={true}>text</Plate>);

            expect(container.firstElementChild).not.toHaveClass('foldable');
        });

        it('should not set `foldable` class with empty `children`', () => {
            const { container } = render(<Plate foldable={true} title='title' />);

            expect(container.firstElementChild).not.toHaveClass('foldable');
        });
    });

    describe('Callbacks tests', () => {
        it('should call `onClick` prop', async () => {
            const cb = jest.fn();
            const dataTestId = 'test-id';
            const { getByTestId } = render(<Plate onClick={cb} dataTestId={dataTestId} />);

            const el = getByTestId(dataTestId);

            fireEvent.click(el);

            expect(cb).toBeCalledTimes(1);
        });

        it('should call `onClose` prop', async () => {
            const cb = jest.fn();
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <Plate hasCloser={true} onClose={cb} dataTestId={dataTestId} />,
            );

            const el = getByTestId(dataTestId);
            const closeEl = el.querySelector('.closer') as Element;

            fireEvent.click(closeEl);

            expect(cb).toBeCalledTimes(1);
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<Plate />);

        expect(unmount).not.toThrowError();
    });
});
