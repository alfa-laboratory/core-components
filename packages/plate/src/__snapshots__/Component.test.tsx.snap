// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plate Snapshots tests should match snapshot 1`] = `
<div>
  <div
    class="component positive"
    role="alert"
    tabindex="-1"
  >
    <div
      class="leftAddons"
    >
      <svg
        fill="currentColor"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zm-1.5-6L17 9.5 15.5 8l-5 5-2-2L7 12.5l3.5 3.5z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="contentContainer"
    >
      <div
        class="title"
      >
        title
      </div>
      <div
        class="content"
      >
        <div
          class="contentInner"
        >
          text
        </div>
      </div>
    </div>
  </div>
</div>
`;
